designer/server/src/main/scala/pl/touk/nussknacker/ui/definition/UIProcessObjectsFactory.scala

```diff
@@ -42,67 +62,119 @@ object UIProcessObjectsFactory {
-      .mapValuesNow(createUIAdditionalPropertyConfig)
-
-    val defaultUseAsyncInterpretationFromConfig =
-      processConfig.as[Option[Boolean]]("asyncExecutionConfig.defaultUseAsyncInterpretation")
-    val defaultAsyncInterpretation: DefaultAsyncInterpretationValue =
-      DefaultAsyncInterpretationValueDeterminer.determine(defaultUseAsyncInterpretationFromConfig)
-
     UIProcessObjects(
       componentGroups = ComponentDefinitionPreparer.prepareComponentsGroupList(
         user = user,
-        processDefinition = uiProcessDefinition,
+        processDefinition = finalProcessDefinition,
+        fragmentInputs = fragmentInputs,
         isFragment = isFragment,
-        componentsConfig = finalComponentsConfig,
-        componentsGroupMapping = componentsGroupMapping,
+        componentsConfig = combinedComponentsConfig,
+        componentsGroupMapping = ComponentsGroupMappingConfigExtractor.extract(modelDataForType.processConfig),
         processCategoryService = processCategoryService,
-        customTransformerAdditionalData = customTransformerAdditionalData,
+        customTransformerAdditionalData = finalProcessDefinition.customStreamTransformers.map {
+          case (idWithName, (_, additionalData)) => (idWithName, additionalData)
+        }.toMap,
         processingType
       ),
-      processDefinition = uiProcessDefinition,
-      componentsConfig = finalComponentsConfig,
-      additionalPropertiesConfig = additionalPropertiesConfigForUi,
+      processDefinition = createUIProcessDefinition(
+        finalProcessDefinition,
+        fragmentInputs,
+        modelDataForType.modelDefinitionWithTypes.typeDefinitions.all.map(prepareClazzDefinition),
+        processCategoryService
+      ),
+      componentsConfig = toComponentsUiConfig(finalProcessDefinition) |+| combinedComponentsConfig,
+      additionalPropertiesConfig = additionalPropertiesConfig
+        .filter(_ =>
+          !isFragment
+        ) // fixme: it should be introduced separate config for additionalPropertiesConfig for fragments. For now we skip that
+        .mapValuesNow(createUIAdditionalPropertyConfig),
       edgesForNodes = ComponentDefinitionPreparer.prepareEdgeTypes(
-        processDefinition = staticObjectsDefinition,
+        processDefinition = finalProcessDefinition,
         isFragment = isFragment,
         fragmentsDetails = fragmentsDetails
       ),
       customActions = deploymentManager.customActions.map(UICustomAction(_)),
-      defaultAsyncInterpretation = defaultAsyncInterpretation.value
+      defaultAsyncInterpretation = getDefaultAsyncInterpretation(modelDataForType.processConfig)
     )
   }
 
+  private def toComponentsUiConfig(
+      processDefinition: ProcessDefinitionWithComponentIds[ObjectDefinition]
+  ): ComponentsUiConfig =
+    processDefinition.allDefinitions.map { case (idWithName, value) => idWithName.name -> value.componentConfig }.toMap
+
+  private def finalizeProcessDefinition(
```

Comment:
The idea of `ObjectDefinition.componentConfig` was to hold a `componentConfig` defined programmatically (in user's `ProcessConfigCreator`) whereas this mentioned `Map` holds a result of combination configs defined programmatically, defined in the static config and now - using this plugin mechanism. But it is minor thing so for me we can stick with this approach 

but ... ;) I see another problem! Using the current approach it is impossible to use `AdditionalComponentsUIConfigProvider` mechanism to provide configuration for fragments and "base" components and it is not acceptable :( I think that we don't need to prepare finalizedProcessDefinition and after that extract config from them but we should think about the general concept, how we want to merge this additional components config provided by `AdditionalComponentsUIConfigProvider` into the existing configuration based on mentioned two sources of components config.

Reply:
you're right that it doesn't allow for configuration of basic components and fragments - I thought this would be ok, especially with fragments getting their own (with same or very similar capabilities) way of editing parameter configs, but perhaps it's an issue

i'll think about a solution

Reply:
The merging logic would be a little bit tricky.  I think that for now, we can leave a comment next to the `AdditionalComponentsUIConfigProvider` that it can provide a configuration only for a standard component. In the future IMO we should return to FE components keyed by `ComponentId` and move the name (label) into `SingleComponentConfig`. You can leave a TODO as well.